pkg/resources/server.go

```diff
@@ -193,7 +202,7 @@ func (rs *resourceServer) PreStartContainer(ctx context.Context,
 func (rs *resourceServer) GetDevicePluginOptions(ctx context.Context, empty *pluginapi.Empty) (*pluginapi.DevicePluginOptions, error) {
 	return &pluginapi.DevicePluginOptions{
 		PreStartRequired:                false,
-		GetPreferredAllocationAvailable: false,
+		GetPreferredAllocationAvailable: (rs.allocator != nil),
```

Comment:
do we really need to set this to true if we end up with the "default" allocator ?

pros to keeping this as is is that we have one flow for allocations.
con is that we add an additional rpc call which is not needed in case of the default allocator

im fine with either, just want to hear other opinions on this one.

Reply:
I think we don't need to set this to true if we doesn't need a specific allocator. Because, as you pointed out, we don't want  extra meaningless rpc call in case of the default allocator.